src/app/session.tsx

```diff
@@ -0,0 +1,47 @@
+import 'server-only';
+import {JWTPayload, SignJWT, jwtVerify} from 'jose';
+// import { SessionPayload } from '@/app/lib/definitions'
+import {cookies} from 'next/headers';
+
+const secretKey = process.env.SESSION_SECRET;
+const encodedKey = new TextEncoder().encode(secretKey);
+
+type SessionPayload = {userId: string; expiresAt: Date};
+
+export async function encrypt(payload: SessionPayload) {
+  return new SignJWT(payload)
+    .setProtectedHeader({alg: 'HS256'})
+    .setIssuedAt()
+    .setExpirationTime('1d')
+    .sign(encodedKey);
+}
+
+export async function decrypt(
+  session: string | undefined = ''
+): Promise<JWTPayload | void> {
+  try {
+    const {payload} = await jwtVerify(session, encodedKey, {
+      algorithms: ['HS256'],
+    });
+    return payload;
+  } catch (error) {
+    console.log('Failed to verify session');
+  }
+}
+
+export async function createSession(userId: string) {
+  const expiresAt = new Date(Date.now() + 1 * 24 * 60 * 60 * 1000);
```

Comment:
we should change the session length to something longer like a week or a month

Reply:
4 weeks works?